```bash
mypy.....................................................................Failed
- hook id: mypy
- duration: 83.98s
- exit code: 1

pandas/core/internals/concat.py:443: error: Incompatible return value type (got "tuple[dtype[Any] | ExtensionDtype, dtype[Any] | ExtensionDtype]", expected "dtype[Any] | ExtensionDtype")  [return-value]
Found 1 error in 1 file (checked 1443 source files)
```

--- a/pandas/core/internals/concat.py
+++ b/pandas/core/internals/concat.py
@@ -440,7 +440,7 @@ def _get_empty_dtype(join_units: Sequence[JoinUnit]) -> DtypeObj:
     """
     if lib.dtypes_all_equal([ju.block.dtype for ju in join_units]):
         empty_dtype = join_units[0].block.dtype
-        return empty_dtype, empty_dtype
+        return empty_dtype
 
     has_none_blocks = any(unit.block.dtype.kind == "V" for unit in join_units)
 

